spark_auto_mapper_fhir/resources/immunization.py

```diff
@@ -1,30 +1,172 @@
+    ImmunizationProgramElegibilityCode,
+)
+from spark_auto_mapper_fhir.valuesets.immunization_reason_code import (
+    ImmunizationReasonCode,
+)
+from spark_auto_mapper_fhir.valuesets.immunization_route import ImmunizationRouteCode
+from spark_auto_mapper_fhir.valuesets.immunization_site import ImmunizationSiteCode
+
+from spark_auto_mapper_fhir.valuesets.immunization_status_reason import (
+    ImmunizationStatusReasonCode,
+)
+from spark_auto_mapper_fhir.complex_types.identifier import Identifier
+from spark_auto_mapper_fhir.valuesets.immunization_subpotent_reason import (
+    ImmunizationSubpotentReasonCode,
+)
+from spark_auto_mapper_fhir.valuesets.immunization_target_disease import (
+    ImmunizationTargetDiseaseCode,
+)
+from spark_auto_mapper_fhir.valuesets.vaccine_code import VaccineCode
 
 
 class Immunization(FhirResourceBase):
     def __init__(
         self,
+        *,
         id_: Optional[FhirId] = None,
         meta: Optional[Meta] = None,
         extension: Optional[FhirList[Extension]] = None,
+        identifier: Optional[FhirList[Identifier]] = None,
+        status: ImmunizationSiteCode,
+        statusReason: Optional[CodeableConcept[ImmunizationStatusReasonCode]] = None,
+        vaccineCode: CodeableConcept[VaccineCode],
+        patient: Reference[Patient],
+        encounter: Optional[Reference[Encounter]] = None,
+        occurrenceDateTime: FhirDateTime,
+        occurrenceString: FhirString,
+        recorded: Optional[FhirDateTime] = None,
+        primarySource: Optional[FhirBoolean] = None,
+        reportOrigin: Optional[CodeableConcept[ImmunizationOriginCode]] = None,
+        location: Optional[Reference[Location]] = None,
+        manufacturer: Optional[Reference[Organization]] = None,
+        lotNumber: Optional[FhirString] = None,
+        expirationDate: Optional[FhirDate] = None,
+        site: Optional[CodeableConcept[ImmunizationSiteCode]] = None,
+        route: Optional[CodeableConcept[ImmunizationRouteCode]] = None,
+        doseQuantity: Optional[SimpleQuantity] = None,
+        performer: Optional[FhirList[ImmunizationFunctionCode]] = None,
+        note: Optional[Annotation] = None,
+        reasonCode: Optional[CodeableConcept[ImmunizationReasonCode]] = None,
+        reasonReference: Optional[
+            Reference[Union[Condition, Observation, "DiagnosticReport"]]
+        ] = None,
+        subpotentReason: Optional[
+            CodeableConcept[ImmunizationSubpotentReasonCode]
+        ] = None,
+        education: Optional[ImmunizationEducationBackboneElement] = None,
+        programEligibility: Optional[
+            CodeableConcept[ImmunizationProgramElegibilityCode]
+        ] = None,
+        fundingSource: Optional[CodeableConcept[ImmunizationFundingSourceCode]] = None,
+        reaction: Optional[ImmunizationReactionBackboneElement] = None,
+        protocolApplied: Optional[ImmunizationTargetDiseaseCode] = None,
```

Comment:
protocolApplied should be a backbone element

Reply:
done